docs-chef-io/content/server/upgrade_server_ha_v2.md

```diff
@@ -163,13 +163,162 @@ Upgrading from Chef Backend 1.x to Chef Backend 2.x requires full cluster downti
+
+    ```ruby
+    chef-backend-ctl cluster-status
+    ```
+
+    ```output
+    Name       IP              GUID                              Role      PG        ES
+    backend-1  192.168.33.215  dc0c6ea77a751f94037cd950e8451fa3  leader    leader    not_master
+    backend-2  192.168.33.216  008782c59d3628b6bb7f43556ac0c66c  follower  follower  not_master
+    backend-3  192.168.33.217  1af654172b1830927a571d9a5ba7965b  follower  follower  master
+    ```
+
+3. Install the new Chef Backend package on each of the server nodes in the cluster:
+
+    RHEL and Centos:
+
+    ```bash
+    yum install PATH_TO_FILE.rpm
+    ```
+
+    Debian and Ubuntu:
+
+    ```bash
+    dpkg -i PATH_TO_FILE.deb
+    ```
+
+4. Next, run the upgrade subcommand on each of the follower nodes:
+
+    ```bash
+    chef-backend-ctl upgrade
+    ```
+
+5. After finishing the upgrades on the follower nodes, **then** upgrade the leader node, using  the `--failover` option:
+
+    ```bash
+    chef-backend-ctl upgrade --failover
+    ```
+
+6. Execute each of these three commands on every Chef Backend server node:
+
+   ```bash
+   sudo chef-backend-ctl stop elasticsearch
+   sudo rm -fr /var/opt/chef-backend/elasticsearch/data/*
+   sudo chef-backend-ctl start elasticsearch
+   ```
+
+   These commands ensure that Chef Backend creates new indices on the latest version installed Elasticsearch version.
+
+7. Identify a front-end node that does not serve requests. You can remove it from the the load balancer or put in maintenance mode.
+
+   Run reconfigure to recreate the index definitions. Reconfigure can be run either when the services are stopped or server is in maintenance mode.
+
+   ```bash
+   sudo chef-server-ctl reconfigure
+   ```
+
+   Alternatively, if the front-end node is not out of the load balancer, On frontends running Infra Server 14.6.32 or later if chef-server-ctl status is running all services you can disable the API temporarily from remote access by running the following command.
+
+   ```bash
+   sudo chef-server-ctl maintenance on
+   ```
+
+   Test the new index by reindexing a single org
+
+   ```bash
+   sudo chef-server-ctl reindex ORGNAME
+
+   sudo /opt/opscode/bin/knife search node *:* -c /etc/opscode/pivotal.rb --server-url https://127.0.0.1:443/organizations/ORGNAME
+   ```
+
+   If successful reindex all organizations
+   
+   We estimate the reindexing operation will take 2 minutes for each 1000 nodes, but the it could take more time, depending on your server hardware and the complexity of your Chef data.(REF:https://docs.chef.io/server/upgrades/#standalone-server)
+
+   ```bash
+   sudo chef-server-ctl reindex -a
+   ```
+
+8. If a frontend was placed in maintenance mode, restore it to regular status
+
+   ```bash
+   sudo chef-server-ctl maintenance off
+   ```
+
+9. Restart the services on the front-end servers
+
+   ```bash
+   chef-server-ctl start
+   ```
+
+### Upgrade Chef Backend Cluster
```

Comment:
Is this section redundant?

Reply:
I see it now--the entire previous section is for systems with earlier installations that upgraded to 2.3.1.